components/brave_today/browser/publishers_controller.cc

```diff
@@ -36,6 +38,45 @@
 
 namespace brave_news {
 
+namespace {
+mojom::Publisher* FindMatchPreferringLocale(
+    const Publishers& publishers,
+    const std::string& preferred_locale,
+    base::RepeatingCallback<bool(const mojom::Publisher&)> matcher) {
+  if (publishers.empty()) {
+    return nullptr;
+  }
+
+  mojom::Publisher* match = nullptr;
+  for (const auto& it : publishers) {
+    auto* publisher = it.second.get();
+    if (!matcher.Run(*publisher)) {
+      continue;
+    }
+
+    auto locale_it =
+        std::find_if(publisher->locales.begin(), publisher->locales.end(),
```

Comment:
optional: base::ranges::find_if would be nice.

```suggestion
        base::ranges::find_if(publisher->locales,
```

Reply:
Or, you might be able to remove lambda by using projection.

```suggestion
        base::ranges::find(publisher->locales, preferred_locale, &mojom::LocaleInfo::locale);
```

Reply:
Wow, with the projection is awesome! Thanks @sangwoo108!